packages/ag-charts-community/src/chart/series/seriesEvents.ts

```diff
@@ -0,0 +1,21 @@
+import type { DataModel, ProcessedData, PropertyDefinition } from '../data/dataModel';
+
+export type SeriesEventType = 'data-update' | 'processData-prerequest' | 'visibility-changed';
+
+export interface BaseSeriesEvent<_T extends SeriesEventType> {}
+
+export interface SeriesDataUpdateEvent extends BaseSeriesEvent<'data-update'> {
+    dataModel: DataModel<any, any, any>;
+    processedData: ProcessedData<any>;
+}
+
+export interface SeriesPrerequestDataEvent extends BaseSeriesEvent<'processData-prerequest'> {
+    props: PropertyDefinition<string>[];
+    isContinuousX: boolean;
+    isContinuousY: boolean;
+}
+
+export interface SeriesVisibilityEvent extends BaseSeriesEvent<'visibility-changed'> {
+    itemId: any;
```

Comment:
Can we add a stricted typing here?

Reply:
I would like to, but this interface needs to be compatible with the parameters of [Series.toggleSeriesItem](https://github.com/ag-grid/ag-charts/blob/ce7c48ca175ae05309f50e8add6761b0b6f7f499/packages/ag-charts-community/src/chart/series/series.ts#L785).

I'm not sure what to use instead. Open to suggestions.